### PR TITLE
Optionally persist the result of a job

### PR DESCRIPTION
**Pull request by ochrstn**
Coming from BullMQ I was missing the ability to persist the job result.

I added the feature as an option that is disabled by default.

### CODE REVIEW

**Comment by ochrstn** [comment]
I wanted to ask if this is going to be reviewed or how I should proceed with my merge request.

**Comment by harisvsulaiman** [comment]
@ochrstn Sorry for the delay. Let me have a look at it

File: lib/agenda/create.ts [hunk 1]
```diff
@@ -16,6 +16,7 @@ export const create = function (this: Agenda, name: string, data: any): Job {
   const priority = this._definitions[name]
     ? this._definitions[name].priority
     : 0;
-  const job = new Job({ name, data, type: "normal", priority, agenda: this });
+  const saveResult = this._definitions[name] ? this._definitions[name].saveResult || false : false
```
Comment (harisvsulaiman) [inline lib/agenda/create.ts:19]: Renaming `saveResult` to `shouldSaveResult` will help conveying that its a boolean.

**Review by harisvsulaiman — APPROVED** [review]

**Comment by harisvsulaiman** [comment]
@koresar can you review this too?

**Review by koresar — CHANGES_REQUESTED** [review]
Please document the new feature in the README.md

**Comment by harisvsulaiman** [comment]
@ochrstn we would appreciate it if you could update the readme.md to document the new changes .
I'm hoping to get this into the agenda 5.0 release.

**Comment by ochrstn** [comment]
> @ochrstn we would appreciate it if you could update the readme.md to document the new changes .
> I'm hoping to get this into the agenda 5.0 release.

> Please document the new feature in the README.md

Sorry for the late response. I will do it this weekend.

**Comment by harisvsulaiman** [comment]
Looks good to me!

**Comment by harisvsulaiman** [comment]
@ochrstn the checks are failing. You will have incorporate the latest master

**Review by harisvsulaiman — APPROVED** [review]

**Comment by harisvsulaiman** [comment]
@ochrstn Thank you for your contribution. please join the agenda slack if you haven't joined already.

**Comment by GW00D** [comment]
Hi, thanks for working on this, quite keen for this feature. Do we know which release this is targeted for and when that might be roughly?

**Comment by koresar** [comment]
We don't know when. It depends on too many people.
But we hope to have this in the next major 5.0.

**Comment by kubanac95** [comment]
Any news on this feature? Seeing it on master I thought it's part of 4.2.1, but looks like it has not aired yet

**Comment by JonLopezGarcia** [comment]
I see that persisting the result of a job run would override the result of the previous job run. 
Is this the intended behavior ? What about persisting the result of each run ? What would be the best way to achieve this ?

**Pull request merged**